fastlane_core/lib/fastlane_core/device_manager.rb

```diff
@@ -40,6 +40,7 @@ def simulators(requested_os_type = "")
         end
 
         output.split(/\n/).each do |line|
+          break if line =~ /^-- Unavailable/
```

Comment:
This is assuming that `-- Unavailable` will always be in the bottom of the output, correct? If for some reason that `-- Unavailable` is not at the bottom of the output, this will break out of this loop completely and skip over some devices, correct? I think this is what your comment below meant of...

>Optionally, I could just never initialize any simulator in a loop unless previously observed os_version was valid.

I think if we could make this a bit safer and not break out that would be ideal. I can see this leading to some hard to debug/discover issues in the future if that `--Unavailable` wasn't always at the bottom.

Thoughts?

Reply:
I updated PR as suggested